hazelcast-client/src/main/java/com/hazelcast/client/spi/ClientProxy.java

```diff
@@ -98,16 +98,13 @@ public final void destroy() {
             context.removeProxy(this);
             try {
                 new ClientInvocation(getClient(), clientMessage).invoke().get();
+                postDestroy();
```

Comment:
I have just wanted it called only when invocation is successful. We may introduce another hook method for error case like `onDestroyError(Exception e)`